signing/src/test/java/tech/pegasys/web3signer/signing/FileValidatorManagerTest.java

```diff
@@ -0,0 +1,105 @@
+import java.util.concurrent.Future;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.tuweni.bytes.Bytes;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class FileValidatorManagerTest {
+  private static final Bytes SALT =
+      Bytes.fromHexString("0x9ac471d9d421bc06d9aefe2b46cf96d11829c51e36ed0b116132be57a9f8c22b");
+  private static final Bytes IV = Bytes.fromHexString("0xcca2c67ec95a1dd13edd986fea372789");
+  private static final BLSKeyPair BLS_KEY_PAIR = BLSTestUtil.randomKeyPair(1);
+
+  @Mock private ArtifactSignerProvider artifactSignerProvider;
+  @Mock private KeystoreFileManager keystoreFileManager;
+
+  @Test
+  @SuppressWarnings("unchecked")
+  void addsValidator() throws IOException, ExecutionException, InterruptedException {
+    final Future<Void> futureAddSigner = Mockito.mock(Future.class);
+    when(artifactSignerProvider.addSigner(any())).thenReturn(futureAddSigner);
+
+    final String keystoreJson = createKeystoreString();
+
+    final FileValidatorManager fileValidatorManager =
+        new FileValidatorManager(
+            artifactSignerProvider, keystoreFileManager, YamlSignerParser.YAML_MAPPER);
+    fileValidatorManager.addValidator(
+        BLS_KEY_PAIR.getPublicKey().toBytesCompressed(), keystoreJson, "password");
+
+    verify(keystoreFileManager)
+        .createKeystoreFiles(
+            eq(BLS_KEY_PAIR.getPublicKey().toString()), eq(keystoreJson), eq("password"));
+    verify(futureAddSigner).get();
+  }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  void deletesValidator() throws ExecutionException, InterruptedException {
+    final Future<Void> futureDeleteSigner = Mockito.mock(Future.class);
+    when(artifactSignerProvider.removeSigner(any())).thenReturn(futureDeleteSigner);
+
+    final FileValidatorManager fileValidatorManager =
+        new FileValidatorManager(
+            artifactSignerProvider, keystoreFileManager, YamlSignerParser.YAML_MAPPER);
+    fileValidatorManager.deleteValidator(BLS_KEY_PAIR.getPublicKey().toBytesCompressed());
+
+    verify(artifactSignerProvider).removeSigner(eq(BLS_KEY_PAIR.getPublicKey().toString()));
+    verify(futureDeleteSigner).get();
+  }
+
+  private String createKeystoreString() throws JsonProcessingException {
+    final Cipher cipher = new Cipher(CipherFunction.AES_128_CTR, IV);
+    final Pbkdf2Param pbkdf2Param = new Pbkdf2Param(32, 262144, HMAC_SHA256, SALT);
+    final KeyStoreData keyStoreData =
+        KeyStore.encrypt(
+            BLS_KEY_PAIR.getSecretKey().toBytes(),
+            BLS_KEY_PAIR.getPublicKey().toBytesCompressed(),
+            "password",
+            "",
+            pbkdf2Param,
+            cipher);
+
+    return new ObjectMapper()
```

Comment:
nit: It might be a good idea to use the global ObjectMapper (if it is exposed). Or at minimum, consider declaring it as static constant.

Reply:
added a static constant